src/reference/02-DetailTopics/02-Configuration/15-In-Process-Classloaders.md

```diff
@@ -0,0 +1,92 @@
+---
+out: In-Process-Classloaders.html
+---
+
+  [Running Project Code]: Running-Project-Code.html
+  [Forking]: Forking.html
+
+
+### In process class loading
+
+When evaluating the `run` and `test` tasks without [forking][Forking], sbt
+creates a custom classloader for each task evaluation run. For many projects, a
+large amount of time can be spent loading the classes for the task. This is
+especially true for projects with many dependencies because the jvm must find
+all of the required classes in a lengthy list of jar files. Even with ssds, the
+io required to locate all of the classes and read them into memory can delay the
+startup of the task by seconds. To reduce this overhead, sbt creates layered
+classloaders where certain layers may be cached between task evaluation runs.
+The inital classloading overhead is a fixed cost, but the startup delay of
+subsequent runs in an interactive session can be reduced by caching the
+intermediate layers.
+
+Another benefit of the layered classloaders is that the jvm may choose to jit
+some of the cached classes. As a result, the total runtime of the task may
+decrease as the interactive session proceeds.
+
+
+### ClassLoader layering strategy
+As of 1.3.0, sbt allows users to select one of three classloader layering
+strategies[*]. The strategy may be set using the `classLoaderLayeringStrategy`
+setting. For example, to use the [AllLibraryJars] strategy in the test
+configuration, set
+
+```
+Test / classLoaderLayeringStrategy := ClassLoaderLayeringStrategy.AllLibraryJars
+```
+
+Descriptions of the strategies follow.
+
+#### ScalaLibrary
+
+Prior to sbt 1.3.0, this was the only ClassLoader layering strategy. The top
+layer of this ClassLoader is able to load the scala standard library classes,
```

Comment:
> The top layer of this ClassLoader

In terms of what's top/bottom, I think it's more intuitive for most users if the upper layers extend lower layers as typically stacked diagrams are written. This is rather unfortunate since "top ClassLoader" means something, but since we are discussing layering strategy, I propose that we flip up/down here.

> The bottom layer loads the Scala standard library classes, the sbt test interface classes, and the Java library.